### PR TITLE
Added a variable to specify a private zone for the route53

### PR DESCRIPTION
**Pull request by ramakantar**
Added a variable to specify a private zone for the route53 zone

### CODE REVIEW

File: modules/lb/variables.tf [hunk 1]
```diff
@@ -31,6 +31,12 @@ variable "domain" {
   description = "this is used to find an existing issued wildcard cert and route53 zone"
 }
 
+variable "private_zone" {
+  type        = "string"
+  description = "this is used to if route53 zone is private or not"
```
Comment (evanphx) [inline modules/lb/variables.tf]: Could you change this to `Set to true if route53 zone is private`?

File: variables.tf [hunk 2]
```diff
@@ -12,6 +12,12 @@ variable "domain" {
   description = "Route53 Domain to manage DNS under"
 }
 
+variable "private_zone" {
+  type        = "string"
+  description = "this is used to if route53 zone is private or not"
```
Comment (evanphx) [inline variables.tf]: Ditto with the other description, could you change it?

File: modules/lb/variables.tf (hunk 1 above)
Comment (ramakantar) [reply modules/lb/variables.tf]: Updated it

File: variables.tf (hunk 2 above)
Comment (ramakantar) [reply variables.tf]: here too

**Comment by ed-brex** [comment]
Likely outside the scope of the PR, but would it make sense for `name` be replaced with `zone_id` (which may also eliminate the need for `update_route53`)?
`var.private_zone` helps the private zone issue, but doesn't cover the case when the zone's `name != var.domain`. I believe an explicit `zone_id` eliminates the need for `private_zone` and if being passed explicitly implies `update_route53`. 
I can see a case where a delegated zone `tfe.example.com` uses an ACM cert `tfe.example.com` which cannot be updated when `domain` must be set to `example.com` and `hostname` (subdomain?) would need to be set to `tfe`

**Comment by bnferguson** [comment]
Closing as stale. The use cases brought up here are something we'd like to address in the future with more composable submodules. Using route53, public or private or not at all should be possible then.

**Pull request closed without merging**